index.ts

```diff
@@ -549,10 +545,38 @@ class ToucanClient {
   };
 
   // --------------------------------------------------------------------------------
+  //  Internal methods
   // --------------------------------------------------------------------------------
-  //  Subgraph related methods
-  // --------------------------------------------------------------------------------
-  // --------------------------------------------------------------------------------
+
+  /**
+   *
+   * @description gets the contract of a pool token based on the symbol
+   * @param pool symbol of the pool (token) to use
+   * @returns a ethers.contract to interact with the pool
+   */
+  private getPoolContract = (pool: poolSymbol): IToucanPoolToken => {
+    return pool == "BCT" ? this.bct : this.nct;
+  };
+}
+
+export class SubgraphInteractions {
+  network: Network;
+  addresses: IfcOneNetworksAddresses;
```

Comment:
can you explain what this interface does? I can't really understand the name and don't know what it's for ✌️

Reply:
ah, got it! I'd suggest `INetworkAddresses`, `INetworkContractAddresses` or `IContractAddressesForNetwork` and thusly `IAddresses` or `IAllAddresses` :)